File: Counting Elements/PermCheck.cpp
// you can use includes, for example:
// #include <algorithm>
#include <set>
// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++11 (g++ 4.8.2)
    int len = A.size();
    int ret = 0;
    if (len == 0)
        return 0;
    set<int> B{};
    for (int i: A)
    {
        if ( i>0 && i<= len )
        {
            B.insert(i);
        }
    }
    if (int(B.size()) == len)
    {
        ret = 1;
    }
    return ret;
}

File: Counting Elements/PermCheck.py
# you can write to stdout for debugging purposes, e.g.
# print "this is a debug message"

def solution(A):
    # write your code in Python 2.7
    from sets import Set
    B=Set()
    len_a = len(A)
    if len_a == 0:
        return 0
    for i in A:
        if i>0 and i <=len_a:
            B.add(i)
    ret = 0
    if len_a == len(B):
        ret = 1
    return ret